tests/run/threading_stress_tests.pyx

```diff
@@ -0,0 +1,56 @@
+# mode: run
```

Comment:
This isn't a brilliant test, in that I couldn't actually get it to fail even without the patch.

I can see the issue in thread sanitizer without though, and can't see the issue in thread sanitizer with the patch. But I don't think it's time to add thread sanitizer to the test suite yet.